src/elli_request.erl

```diff
@@ -292,8 +290,15 @@ remove_whitespace(Bin) ->
 %% @doc Serialize the `Req'uest record to a proplist.
 %% Useful for logging.
 -spec to_proplist(elli:req()) -> proplists:proplist().
-to_proplist(#req{} = Req) ->
-    lists:zip(record_info(fields, req), tl(tuple_to_list(Req))).
+to_proplist(#{} = Req) ->
+    Defaults = #{scheme => undefined,
+                 host   => undefined,
+                 port   => undefined,
+                 path   => [],
+                 args   => [],
+                 socket => undefined
+                },
+    maps:to_list(maps:merge(Defaults, Req)).
```

Comment:
There isn't, except if always using `maps:get` with a default.

Part of why it may be better to stick with a record. I'm undecided on this. Are there specific benefits to using a map for req?

Reply:
Not that I can think of, except that it'd be easier to do ad hoc extensions à la [Ring](https://github.com/ring-clojure/ring) then.